### DISCUSSION TITLE
Question about cleanup method - APS v4

### DISCUSSION THREAD

**Discussion by mattewid** [agronholm/apscheduler#958] [Q&A]
Hello,

My question is about the DataStore cleanup method. Exactly in this part of release jobs: https://github.com/agronholm/apscheduler/blob/8ed2b1babf6c7c173f2c8ed11b9417c891aa0dac/src/apscheduler/datastores/mongodb.py#L861

My question is, Even if the jobs are running, will they be released/deleted?

**Comment by agronholm** [comment]
They shouldn't be, as schedulers would extend those leases long before they expire.

**Reply by mattewid** [reply]
Got it,

In the Scheduler, if the **cleanup_interval** parameter is set to be shorter than the **lease_duration** parameter, do you agree that the automatic cleanup will release/delete jobs that are still running? _Analyzing the code, it seems that this is what I understood. I just want to clarify this doubt._

**Reply by agronholm** [reply]
Given that schedulers automatically extend their leases when half of the lease duration has elapsed, can you point out how that would happen? [Here](https://github.com/agronholm/apscheduler/blob/master/src/apscheduler/_schedulers/async_.py#L933) is the relevant code.
